OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -492,8 +492,11 @@ void ProtectOwn(IBot bot, Actor attacker)
 		{
 			foreach (var s in Squads)
 			{
-				if (s.IsValid && FindEnemies([attacker], s.Units.First()).Any()
-					&& (s.Units.First().Location - attacker.Location).LengthSquared <= Info.ProtectUnitScanRadius * Info.ProtectUnitScanRadius)
+				var livingUnitInSquad = s.Units.FirstOrDefault(x => !x.IsDead);
+				if (livingUnitInSquad == null)
+					continue;
+				if (s.IsValid && FindEnemies([attacker], livingUnitInSquad).Any()
```

Comment:
seems a newline is missing. did you run a `make check`?

wouldn't it make more sense to still do the `s.IsValid` check first, to avoid the search if not valid?

Reply:
Maybe remove the invalid actor is better, like what it is doing in `protectSq.Units.RemoveWhere(unitCannotBeOrdered);`